tests/testthat/test_gdp_helpers.R

```diff
@@ -0,0 +1,39 @@
+## Test the gdp join function
+
+context('gdp-join')
+library('tibble', warn.conflicts=FALSE)
+library('magrittr', warn.conflicts=FALSE)
+
+## set up test data
+past <- tibble(year=c(2000,2005,2010)) %>%
+  tidyr::crossing(iso=c('xxl', 'xxp'))  %>%
+  dplyr::mutate(gdp=if_else(iso=='xxl', 10*year, 5*year))
+
+future <- tibble(year=rep(c(2005, 2010, 2015), 2), iso=c(rep('xxl',3), rep('xxp',3)),
+                 gdp=c(cumprod(rep(1.05,3)), cumprod(rep(1.02,3)) ) ) %>%
+  tidyr::crossing(scenario=c('gud','bad')) %>%
+  dplyr::mutate(gdp=if_else(scenario == 'gud', gdp, gdp*0.9^((year-2005)/5)))
+
+test_that('Joining time series with future scenarios works', {
+  rslt <- gcamdata:::join.gdp.ts(past, future, 'iso')
```

Comment:
Weird.  I could have sworn that this one failed for me when I first tried it without the qualifier.  Maybe I forgot to reload the package after adding the function and when I changed the test I reloaded then.

Anyway: ✅